src/init.cpp

```diff
@@ -1434,6 +1430,11 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
     };
     ApplyArgsManOptions(args, mempool_opts);
     mempool_opts.check_ratio = std::clamp<int>(mempool_opts.check_ratio, 0, 1000000);
+
+    size_t descendant_limit_bytes = mempool_opts.limits.descendant_size_vbytes * 40;
+    if (mempool_opts.max_size_bytes < 0 || mempool_opts.max_size_bytes < descendant_limit_bytes) {
```

Comment:
7e90c69f1be6f8ba52e1f24b983379d00aad2f65:

this doesn't work. You can't parse a signed integer and assign it to an unsigned type, then use `< 0` to check if it originally was negative.

Reply:
Already reported here : https://github.com/bitcoin/bitcoin/pull/25290#discussion_r902827767, get caught as a compilation warning.

Reply:
To solve, it would either need to be cast back with `int64_t(...) < 0` or the type changed.